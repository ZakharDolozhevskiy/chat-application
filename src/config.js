export const api = {
  host: 'http://185.13.90.140:8081/',
  channel: 'message'
};

export const routes = {
  MAIN: '/',
  SETTINGS: '/settings'
};