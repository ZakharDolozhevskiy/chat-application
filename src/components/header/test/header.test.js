import React from 'react';
import toJson from 'enzyme-to-json';
import { Header } from '../index';
import { mount } from 'enzyme';

import { routes } from '../../../config';

jest.mock('../../../translator/translator-context');
jest.mock('react-router-dom', () => ({ NavLink: props => props.children }));

describe('<Header/>', () => {
  const component = mount(
    <Header path={routes.MAIN} className="" counter={0}/>
  );

  it('should render header component', () => {
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('.badge').length).toBe(0);
  });

  it('should render header with unread messages badge', () => {
    component.setProps({ path: routes.SETTINGS, counter: 10 });

    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('.badge').length).not.toBe(0);
  });
});