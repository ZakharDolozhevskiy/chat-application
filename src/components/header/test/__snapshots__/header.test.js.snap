// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render header component 1`] = `
<Header
  className=""
  counter={0}
  path="/"
>
  <Consumer>
    <WithStyles(AppBar)
      className=""
      color="inherit"
      position="static"
    >
      <AppBar
        className=""
        classes={
          Object {
            "colorDefault": "MuiAppBar-colorDefault-6",
            "colorPrimary": "MuiAppBar-colorPrimary-7",
            "colorSecondary": "MuiAppBar-colorSecondary-8",
            "positionAbsolute": "MuiAppBar-positionAbsolute-3",
            "positionFixed": "MuiAppBar-positionFixed-2",
            "positionStatic": "MuiAppBar-positionStatic-5",
            "positionSticky": "MuiAppBar-positionSticky-4",
            "root": "MuiAppBar-root-1",
          }
        }
        color="inherit"
        position="static"
      >
        <WithStyles(Paper)
          className="MuiAppBar-root-1 MuiAppBar-positionStatic-5"
          component="header"
          elevation={4}
          square={true}
        >
          <Paper
            className="MuiAppBar-root-1 MuiAppBar-positionStatic-5"
            classes={
              Object {
                "elevation0": "MuiPaper-elevation0-11",
                "elevation1": "MuiPaper-elevation1-12",
                "elevation10": "MuiPaper-elevation10-21",
                "elevation11": "MuiPaper-elevation11-22",
                "elevation12": "MuiPaper-elevation12-23",
                "elevation13": "MuiPaper-elevation13-24",
                "elevation14": "MuiPaper-elevation14-25",
                "elevation15": "MuiPaper-elevation15-26",
                "elevation16": "MuiPaper-elevation16-27",
                "elevation17": "MuiPaper-elevation17-28",
                "elevation18": "MuiPaper-elevation18-29",
                "elevation19": "MuiPaper-elevation19-30",
                "elevation2": "MuiPaper-elevation2-13",
                "elevation20": "MuiPaper-elevation20-31",
                "elevation21": "MuiPaper-elevation21-32",
                "elevation22": "MuiPaper-elevation22-33",
                "elevation23": "MuiPaper-elevation23-34",
                "elevation24": "MuiPaper-elevation24-35",
                "elevation3": "MuiPaper-elevation3-14",
                "elevation4": "MuiPaper-elevation4-15",
                "elevation5": "MuiPaper-elevation5-16",
                "elevation6": "MuiPaper-elevation6-17",
                "elevation7": "MuiPaper-elevation7-18",
                "elevation8": "MuiPaper-elevation8-19",
                "elevation9": "MuiPaper-elevation9-20",
                "root": "MuiPaper-root-9",
                "rounded": "MuiPaper-rounded-10",
              }
            }
            component="header"
            elevation={4}
            square={true}
          >
            <header
              className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5"
            >
              <WithStyles(Toolbar)>
                <Toolbar
                  classes={
                    Object {
                      "gutters": "MuiToolbar-gutters-37",
                      "root": "MuiToolbar-root-36",
                    }
                  }
                  disableGutters={false}
                >
                  <div
                    className="MuiToolbar-root-36 MuiToolbar-gutters-37"
                  >
                    <NavLink
                      className="nav-link"
                      exact={true}
                      to="/"
                    >
                      <WithStyles(Button)
                        color="inherit"
                      >
                        <Button
                          classes={
                            Object {
                              "colorInherit": "MuiButton-colorInherit-57",
                              "contained": "MuiButton-contained-47",
                              "containedPrimary": "MuiButton-containedPrimary-48",
                              "containedSecondary": "MuiButton-containedSecondary-49",
                              "disabled": "MuiButton-disabled-56",
                              "extendedFab": "MuiButton-extendedFab-54",
                              "fab": "MuiButton-fab-53",
                              "flat": "MuiButton-flat-43",
                              "flatPrimary": "MuiButton-flatPrimary-44",
                              "flatSecondary": "MuiButton-flatSecondary-45",
                              "focusVisible": "MuiButton-focusVisible-55",
                              "fullWidth": "MuiButton-fullWidth-61",
                              "label": "MuiButton-label-39",
                              "mini": "MuiButton-mini-58",
                              "outlined": "MuiButton-outlined-46",
                              "raised": "MuiButton-raised-50",
                              "raisedPrimary": "MuiButton-raisedPrimary-51",
                              "raisedSecondary": "MuiButton-raisedSecondary-52",
                              "root": "MuiButton-root-38",
                              "sizeLarge": "MuiButton-sizeLarge-60",
                              "sizeSmall": "MuiButton-sizeSmall-59",
                              "text": "MuiButton-text-40",
                              "textPrimary": "MuiButton-textPrimary-41",
                              "textSecondary": "MuiButton-textSecondary-42",
                            }
                          }
                          color="inherit"
                          component="button"
                          disableFocusRipple={false}
                          disabled={false}
                          fullWidth={false}
                          mini={false}
                          size="medium"
                          type="button"
                          variant="text"
                        >
                          <WithStyles(ButtonBase)
                            className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                            component="button"
                            disabled={false}
                            focusRipple={true}
                            focusVisibleClassName="MuiButton-focusVisible-55"
                            type="button"
                          >
                            <ButtonBase
                              centerRipple={false}
                              className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                              classes={
                                Object {
                                  "disabled": "MuiButtonBase-disabled-63",
                                  "focusVisible": "MuiButtonBase-focusVisible-64",
                                  "root": "MuiButtonBase-root-62",
                                }
                              }
                              component="button"
                              disableRipple={false}
                              disableTouchRipple={false}
                              disabled={false}
                              focusRipple={true}
                              focusVisibleClassName="MuiButton-focusVisible-55"
                              tabIndex="0"
                              type="button"
                            >
                              <button
                                className="MuiButtonBase-root-62 MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                                disabled={false}
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseDown={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                tabIndex="0"
                                type="button"
                              >
                                <span
                                  className="MuiButton-label-39"
                                >
                                  Chat
                                </span>
                                <WithStyles(TouchRipple)
                                  center={false}
                                  innerRef={[Function]}
                                >
                                  <TouchRipple
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child-69",
                                        "childLeaving": "MuiTouchRipple-childLeaving-70",
                                        "childPulsate": "MuiTouchRipple-childPulsate-71",
                                        "ripple": "MuiTouchRipple-ripple-66",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                                        "root": "MuiTouchRipple-root-65",
                                      }
                                    }
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      className="MuiTouchRipple-root-65"
                                      component="span"
                                      enter={true}
                                      exit={true}
                                    >
                                      <span
                                        className="MuiTouchRipple-root-65"
                                      />
                                    </TransitionGroup>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </button>
                            </ButtonBase>
                          </WithStyles(ButtonBase)>
                        </Button>
                      </WithStyles(Button)>
                    </NavLink>
                    <NavLink
                      className="nav-link"
                      exact={true}
                      to="/settings"
                    >
                      <WithStyles(Button)
                        color="inherit"
                      >
                        <Button
                          classes={
                            Object {
                              "colorInherit": "MuiButton-colorInherit-57",
                              "contained": "MuiButton-contained-47",
                              "containedPrimary": "MuiButton-containedPrimary-48",
                              "containedSecondary": "MuiButton-containedSecondary-49",
                              "disabled": "MuiButton-disabled-56",
                              "extendedFab": "MuiButton-extendedFab-54",
                              "fab": "MuiButton-fab-53",
                              "flat": "MuiButton-flat-43",
                              "flatPrimary": "MuiButton-flatPrimary-44",
                              "flatSecondary": "MuiButton-flatSecondary-45",
                              "focusVisible": "MuiButton-focusVisible-55",
                              "fullWidth": "MuiButton-fullWidth-61",
                              "label": "MuiButton-label-39",
                              "mini": "MuiButton-mini-58",
                              "outlined": "MuiButton-outlined-46",
                              "raised": "MuiButton-raised-50",
                              "raisedPrimary": "MuiButton-raisedPrimary-51",
                              "raisedSecondary": "MuiButton-raisedSecondary-52",
                              "root": "MuiButton-root-38",
                              "sizeLarge": "MuiButton-sizeLarge-60",
                              "sizeSmall": "MuiButton-sizeSmall-59",
                              "text": "MuiButton-text-40",
                              "textPrimary": "MuiButton-textPrimary-41",
                              "textSecondary": "MuiButton-textSecondary-42",
                            }
                          }
                          color="inherit"
                          component="button"
                          disableFocusRipple={false}
                          disabled={false}
                          fullWidth={false}
                          mini={false}
                          size="medium"
                          type="button"
                          variant="text"
                        >
                          <WithStyles(ButtonBase)
                            className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                            component="button"
                            disabled={false}
                            focusRipple={true}
                            focusVisibleClassName="MuiButton-focusVisible-55"
                            type="button"
                          >
                            <ButtonBase
                              centerRipple={false}
                              className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                              classes={
                                Object {
                                  "disabled": "MuiButtonBase-disabled-63",
                                  "focusVisible": "MuiButtonBase-focusVisible-64",
                                  "root": "MuiButtonBase-root-62",
                                }
                              }
                              component="button"
                              disableRipple={false}
                              disableTouchRipple={false}
                              disabled={false}
                              focusRipple={true}
                              focusVisibleClassName="MuiButton-focusVisible-55"
                              tabIndex="0"
                              type="button"
                            >
                              <button
                                className="MuiButtonBase-root-62 MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                                disabled={false}
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseDown={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                tabIndex="0"
                                type="button"
                              >
                                <span
                                  className="MuiButton-label-39"
                                >
                                  Settings
                                </span>
                                <WithStyles(TouchRipple)
                                  center={false}
                                  innerRef={[Function]}
                                >
                                  <TouchRipple
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child-69",
                                        "childLeaving": "MuiTouchRipple-childLeaving-70",
                                        "childPulsate": "MuiTouchRipple-childPulsate-71",
                                        "ripple": "MuiTouchRipple-ripple-66",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                                        "root": "MuiTouchRipple-root-65",
                                      }
                                    }
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      className="MuiTouchRipple-root-65"
                                      component="span"
                                      enter={true}
                                      exit={true}
                                    >
                                      <span
                                        className="MuiTouchRipple-root-65"
                                      />
                                    </TransitionGroup>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </button>
                            </ButtonBase>
                          </WithStyles(ButtonBase)>
                        </Button>
                      </WithStyles(Button)>
                    </NavLink>
                  </div>
                </Toolbar>
              </WithStyles(Toolbar)>
            </header>
          </Paper>
        </WithStyles(Paper)>
      </AppBar>
    </WithStyles(AppBar)>
  </Consumer>
</Header>
`;

exports[`<Header/> should render header with unread messages badge 1`] = `
<Header
  className=""
  counter={10}
  path="/settings"
>
  <Consumer>
    <WithStyles(AppBar)
      className=""
      color="inherit"
      position="static"
    >
      <AppBar
        className=""
        classes={
          Object {
            "colorDefault": "MuiAppBar-colorDefault-6",
            "colorPrimary": "MuiAppBar-colorPrimary-7",
            "colorSecondary": "MuiAppBar-colorSecondary-8",
            "positionAbsolute": "MuiAppBar-positionAbsolute-3",
            "positionFixed": "MuiAppBar-positionFixed-2",
            "positionStatic": "MuiAppBar-positionStatic-5",
            "positionSticky": "MuiAppBar-positionSticky-4",
            "root": "MuiAppBar-root-1",
          }
        }
        color="inherit"
        position="static"
      >
        <WithStyles(Paper)
          className="MuiAppBar-root-1 MuiAppBar-positionStatic-5"
          component="header"
          elevation={4}
          square={true}
        >
          <Paper
            className="MuiAppBar-root-1 MuiAppBar-positionStatic-5"
            classes={
              Object {
                "elevation0": "MuiPaper-elevation0-11",
                "elevation1": "MuiPaper-elevation1-12",
                "elevation10": "MuiPaper-elevation10-21",
                "elevation11": "MuiPaper-elevation11-22",
                "elevation12": "MuiPaper-elevation12-23",
                "elevation13": "MuiPaper-elevation13-24",
                "elevation14": "MuiPaper-elevation14-25",
                "elevation15": "MuiPaper-elevation15-26",
                "elevation16": "MuiPaper-elevation16-27",
                "elevation17": "MuiPaper-elevation17-28",
                "elevation18": "MuiPaper-elevation18-29",
                "elevation19": "MuiPaper-elevation19-30",
                "elevation2": "MuiPaper-elevation2-13",
                "elevation20": "MuiPaper-elevation20-31",
                "elevation21": "MuiPaper-elevation21-32",
                "elevation22": "MuiPaper-elevation22-33",
                "elevation23": "MuiPaper-elevation23-34",
                "elevation24": "MuiPaper-elevation24-35",
                "elevation3": "MuiPaper-elevation3-14",
                "elevation4": "MuiPaper-elevation4-15",
                "elevation5": "MuiPaper-elevation5-16",
                "elevation6": "MuiPaper-elevation6-17",
                "elevation7": "MuiPaper-elevation7-18",
                "elevation8": "MuiPaper-elevation8-19",
                "elevation9": "MuiPaper-elevation9-20",
                "root": "MuiPaper-root-9",
                "rounded": "MuiPaper-rounded-10",
              }
            }
            component="header"
            elevation={4}
            square={true}
          >
            <header
              className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5"
            >
              <WithStyles(Toolbar)>
                <Toolbar
                  classes={
                    Object {
                      "gutters": "MuiToolbar-gutters-37",
                      "root": "MuiToolbar-root-36",
                    }
                  }
                  disableGutters={false}
                >
                  <div
                    className="MuiToolbar-root-36 MuiToolbar-gutters-37"
                  >
                    <NavLink
                      className="nav-link"
                      exact={true}
                      to="/"
                    >
                      <WithStyles(Badge)
                        badgeContent={10}
                        className="badge"
                      >
                        <Badge
                          badgeContent={10}
                          className="badge"
                          classes={
                            Object {
                              "badge": "MuiBadge-badge-73",
                              "colorError": "MuiBadge-colorError-76",
                              "colorPrimary": "MuiBadge-colorPrimary-74",
                              "colorSecondary": "MuiBadge-colorSecondary-75",
                              "root": "MuiBadge-root-72",
                            }
                          }
                          color="default"
                          component="span"
                        >
                          <span
                            className="MuiBadge-root-72 badge"
                          >
                            <WithStyles(Button)
                              color="inherit"
                            >
                              <Button
                                classes={
                                  Object {
                                    "colorInherit": "MuiButton-colorInherit-57",
                                    "contained": "MuiButton-contained-47",
                                    "containedPrimary": "MuiButton-containedPrimary-48",
                                    "containedSecondary": "MuiButton-containedSecondary-49",
                                    "disabled": "MuiButton-disabled-56",
                                    "extendedFab": "MuiButton-extendedFab-54",
                                    "fab": "MuiButton-fab-53",
                                    "flat": "MuiButton-flat-43",
                                    "flatPrimary": "MuiButton-flatPrimary-44",
                                    "flatSecondary": "MuiButton-flatSecondary-45",
                                    "focusVisible": "MuiButton-focusVisible-55",
                                    "fullWidth": "MuiButton-fullWidth-61",
                                    "label": "MuiButton-label-39",
                                    "mini": "MuiButton-mini-58",
                                    "outlined": "MuiButton-outlined-46",
                                    "raised": "MuiButton-raised-50",
                                    "raisedPrimary": "MuiButton-raisedPrimary-51",
                                    "raisedSecondary": "MuiButton-raisedSecondary-52",
                                    "root": "MuiButton-root-38",
                                    "sizeLarge": "MuiButton-sizeLarge-60",
                                    "sizeSmall": "MuiButton-sizeSmall-59",
                                    "text": "MuiButton-text-40",
                                    "textPrimary": "MuiButton-textPrimary-41",
                                    "textSecondary": "MuiButton-textSecondary-42",
                                  }
                                }
                                color="inherit"
                                component="button"
                                disableFocusRipple={false}
                                disabled={false}
                                fullWidth={false}
                                mini={false}
                                size="medium"
                                type="button"
                                variant="text"
                              >
                                <WithStyles(ButtonBase)
                                  className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                                  component="button"
                                  disabled={false}
                                  focusRipple={true}
                                  focusVisibleClassName="MuiButton-focusVisible-55"
                                  type="button"
                                >
                                  <ButtonBase
                                    centerRipple={false}
                                    className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                                    classes={
                                      Object {
                                        "disabled": "MuiButtonBase-disabled-63",
                                        "focusVisible": "MuiButtonBase-focusVisible-64",
                                        "root": "MuiButtonBase-root-62",
                                      }
                                    }
                                    component="button"
                                    disableRipple={false}
                                    disableTouchRipple={false}
                                    disabled={false}
                                    focusRipple={true}
                                    focusVisibleClassName="MuiButton-focusVisible-55"
                                    tabIndex="0"
                                    type="button"
                                  >
                                    <button
                                      className="MuiButtonBase-root-62 MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                                      disabled={false}
                                      onBlur={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchMove={[Function]}
                                      onTouchStart={[Function]}
                                      tabIndex="0"
                                      type="button"
                                    >
                                      <span
                                        className="MuiButton-label-39"
                                      >
                                        Chat
                                      </span>
                                      <WithStyles(TouchRipple)
                                        center={false}
                                        innerRef={[Function]}
                                      >
                                        <TouchRipple
                                          center={false}
                                          classes={
                                            Object {
                                              "child": "MuiTouchRipple-child-69",
                                              "childLeaving": "MuiTouchRipple-childLeaving-70",
                                              "childPulsate": "MuiTouchRipple-childPulsate-71",
                                              "ripple": "MuiTouchRipple-ripple-66",
                                              "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                                              "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                                              "root": "MuiTouchRipple-root-65",
                                            }
                                          }
                                        >
                                          <TransitionGroup
                                            childFactory={[Function]}
                                            className="MuiTouchRipple-root-65"
                                            component="span"
                                            enter={true}
                                            exit={true}
                                          >
                                            <span
                                              className="MuiTouchRipple-root-65"
                                            />
                                          </TransitionGroup>
                                        </TouchRipple>
                                      </WithStyles(TouchRipple)>
                                    </button>
                                  </ButtonBase>
                                </WithStyles(ButtonBase)>
                              </Button>
                            </WithStyles(Button)>
                            <span
                              className="MuiBadge-badge-73"
                            >
                              10
                            </span>
                          </span>
                        </Badge>
                      </WithStyles(Badge)>
                    </NavLink>
                    <NavLink
                      className="nav-link"
                      exact={true}
                      to="/settings"
                    >
                      <WithStyles(Button)
                        color="inherit"
                      >
                        <Button
                          classes={
                            Object {
                              "colorInherit": "MuiButton-colorInherit-57",
                              "contained": "MuiButton-contained-47",
                              "containedPrimary": "MuiButton-containedPrimary-48",
                              "containedSecondary": "MuiButton-containedSecondary-49",
                              "disabled": "MuiButton-disabled-56",
                              "extendedFab": "MuiButton-extendedFab-54",
                              "fab": "MuiButton-fab-53",
                              "flat": "MuiButton-flat-43",
                              "flatPrimary": "MuiButton-flatPrimary-44",
                              "flatSecondary": "MuiButton-flatSecondary-45",
                              "focusVisible": "MuiButton-focusVisible-55",
                              "fullWidth": "MuiButton-fullWidth-61",
                              "label": "MuiButton-label-39",
                              "mini": "MuiButton-mini-58",
                              "outlined": "MuiButton-outlined-46",
                              "raised": "MuiButton-raised-50",
                              "raisedPrimary": "MuiButton-raisedPrimary-51",
                              "raisedSecondary": "MuiButton-raisedSecondary-52",
                              "root": "MuiButton-root-38",
                              "sizeLarge": "MuiButton-sizeLarge-60",
                              "sizeSmall": "MuiButton-sizeSmall-59",
                              "text": "MuiButton-text-40",
                              "textPrimary": "MuiButton-textPrimary-41",
                              "textSecondary": "MuiButton-textSecondary-42",
                            }
                          }
                          color="inherit"
                          component="button"
                          disableFocusRipple={false}
                          disabled={false}
                          fullWidth={false}
                          mini={false}
                          size="medium"
                          type="button"
                          variant="text"
                        >
                          <WithStyles(ButtonBase)
                            className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                            component="button"
                            disabled={false}
                            focusRipple={true}
                            focusVisibleClassName="MuiButton-focusVisible-55"
                            type="button"
                          >
                            <ButtonBase
                              centerRipple={false}
                              className="MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                              classes={
                                Object {
                                  "disabled": "MuiButtonBase-disabled-63",
                                  "focusVisible": "MuiButtonBase-focusVisible-64",
                                  "root": "MuiButtonBase-root-62",
                                }
                              }
                              component="button"
                              disableRipple={false}
                              disableTouchRipple={false}
                              disabled={false}
                              focusRipple={true}
                              focusVisibleClassName="MuiButton-focusVisible-55"
                              tabIndex="0"
                              type="button"
                            >
                              <button
                                className="MuiButtonBase-root-62 MuiButton-root-38 MuiButton-text-40 MuiButton-flat-43 MuiButton-colorInherit-57"
                                disabled={false}
                                onBlur={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyUp={[Function]}
                                onMouseDown={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchMove={[Function]}
                                onTouchStart={[Function]}
                                tabIndex="0"
                                type="button"
                              >
                                <span
                                  className="MuiButton-label-39"
                                >
                                  Settings
                                </span>
                                <WithStyles(TouchRipple)
                                  center={false}
                                  innerRef={[Function]}
                                >
                                  <TouchRipple
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child-69",
                                        "childLeaving": "MuiTouchRipple-childLeaving-70",
                                        "childPulsate": "MuiTouchRipple-childPulsate-71",
                                        "ripple": "MuiTouchRipple-ripple-66",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                                        "root": "MuiTouchRipple-root-65",
                                      }
                                    }
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      className="MuiTouchRipple-root-65"
                                      component="span"
                                      enter={true}
                                      exit={true}
                                    >
                                      <span
                                        className="MuiTouchRipple-root-65"
                                      />
                                    </TransitionGroup>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </button>
                            </ButtonBase>
                          </WithStyles(ButtonBase)>
                        </Button>
                      </WithStyles(Button)>
                    </NavLink>
                  </div>
                </Toolbar>
              </WithStyles(Toolbar)>
            </header>
          </Paper>
        </WithStyles(Paper)>
      </AppBar>
    </WithStyles(AppBar)>
  </Consumer>
</Header>
`;
